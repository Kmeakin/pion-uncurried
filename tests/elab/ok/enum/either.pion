enum Either(A: Type, B: Type) {
    Left(x: A),
    Right(y: B),
}

let l = Left;
let r = Right;