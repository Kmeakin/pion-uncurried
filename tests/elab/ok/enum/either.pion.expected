enum Either(A: Type, B: Type): Type {
    Left(x: A): Either(A, B),
    Right(y: B): Either(A, B),
}

let l: fn(A: Type, B: Type, x: A) -> Either(A, B) = Left;

let r: fn(A: Type, B: Type, y: B) -> Either(A, B) = Right;
